modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -172,7 +192,12 @@ public CompletableFuture<Void> createZone(DistributionZoneConfigurationParameter
                 } catch (IllegalArgumentException e) {
                     throw new DistributionZoneAlreadyExistsException(distributionZoneCfg.name(), e);
                 } catch (Exception e) {
-                    throw new IgniteInternalException(UNEXPECTED_ERR, distributionZoneCfg.name(), e);
+                    throw withCause(
```

Comment:
What's the point in catching an exception and re-throwing IgniteInternalException here?

Reply:
Please add all possible exceptions to the @throws clause explaining why it's happening.

Reply:
fixed